### PR TITLE
Add vector property types

### LINKED ISSUE #71: 🎯 Vector property
> Refer [to the contributor docs](https://bitbra.in/pandora/#/contribute/adding-new-property) to learn how to generally add new properties to Pandora!

A Vector property (which could be Vector2, Vector3 — Vector2i and 3i would be nice but not necessary —) would help a lot.

### PR DESCRIPTION
**Pull request by nkrisc**
## Description

Added four new vector properties:
- Vector2i
- Vector2
- Vector3i
- Vector3

Added related getter and setter methods to PandoraEntity for each new property type.

Updated docs to reflect new properties.

## Addressed issues

- closes https://github.com/bitbrain/pandora/issues/71

## Screenshots
pandora_vector_properties

### CODE REVIEW

File: addons/pandora/ui/components/properties/vector/vector_property.gd [hunk 1]
```diff
@@ -0,0 +1,117 @@
+@tool
+extends PandoraPropertyControl
+
+
+enum VectorType { VECTOR_2, VECTOR_2i, VECTOR_3, VECTOR_3i }
```
Comment (bitbrain) [inline addons/pandora/ui/components/properties/vector/vector_property.gd]: Just for consistency, ensure that enums are always upper-case, e.g. VECTOR_2I

File: addons/pandora/ui/components/properties/vector/vector_property.gd [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+			input.min_value = min_value
+			input.max_value = max_value
+			input.step = step
+			if value[i] != input.value:
+				input.set_value_no_signal(value[i])
+
+
+
+func _setting_changed(key:String) -> void:
+	if (vector_type == VectorType.VECTOR_2 &&
```
Comment (bitbrain) [inline addons/pandora/ui/components/properties/vector/vector_property.gd]: ensure to use brackets, as `&&` has priority over `||`

also, usually replace `&&` with `and` and `||` with `or` for better readability.

**Review by bitbrain — CHANGES_REQUESTED** [review]
@nkrisc tests seem to fail - make sure to check the logs for the errors. You also seem to have introduced a memory leak via stray nodes. Ensure to always free up nodes you instantiate via new() that are objects.

https://mikeschulze.github.io/gdUnit3/advanced_testing/orphan/

**Comment by nkrisc** [comment]
I believe the issues, including the stray nodes, are related to the fact I used the EditorSpinSlider node which fails to instantiate since the tests are not run in the editor and that particular control is only available in the editor itself. The only objects I directly create via `new()` are the EditorSpinSliders, and they are added to the root node of the scene so they should be freed when the property controls are freed. Looking at the logs, it seems the stray nodes occurred during the property bar test, probably when the property scenes are initially being instantiated. But I'll have to look more closely later today.

I used this control since it's an editor plugin and to ensure consistency  with vector input fields elsewhere in the editor.

However, seems using that control is essentially untestable. So the options I see:

1. Just use a SpinBox instead and have it be different.
2. Check if it's running in the editor context or not, and choose SpinBox or EditorSpinSlider based on that
3. Recreate the EditorSpinSlider node (probably not worth it)

**Comment by nkrisc** [comment]
I replaced the EditorSpinSlider with a SpinBox to keep things simple for now and all tests pass on my end. I'll leave that as a possible future enhancement.

I also realized I wasn't using GDUnit correctly initially and that's why I didn't catch it earlier - noob mistake.

**Comment by bitbrain** [comment]
Thank you!

**Review by bitbrain — APPROVED** [review]

**Pull request merged**